packages/server/lib/cloud/api.ts

```diff
@@ -88,18 +87,32 @@ const rp = request.defaults((params: CypressRequestOptions, callback) => {
     if (params.encrypt === true || params.encrypt === 'always') {
       const { secretKey, jwe } = await enc.encryptRequest(params)
 
+      // TODO: double check the logic below here with @tgriesser
       params.transform = async function (body, response) {
-        if (response.headers['x-cypress-encrypted'] || params.encrypt === 'always' && response.statusCode < 500) {
+        const { statusCode } = response
+
+        // response is valid and we are encrypting
+        if (response.headers['x-cypress-encrypted'] || params.encrypt === 'always') {
           let decryptedBody
 
+          // TODO: if body is null/undefined throw a custom error
+
           try {
             decryptedBody = await enc.decryptResponse(body, secretKey)
           } catch (e) {
+            // we failed decrypting the response...
+
+            // if status code is >=500 or 404 just return body
+            if (statusCode >= 500 || statusCode === 404 || statusCode === 422) {
+              return body
+            }
+
             throw new DecryptionError(e.message)
           }
 
           // If we've hit an encrypted payload error case, we need to re-constitute the error
           // as it would happen normally, with the body as an error property
+          // TODO: need to look harder at this to better understand why its necessary
```

Comment:
You'll see in the system tests if you remove this `if` block why it's needed. I believe it's because without it you'll get the original body in the `StatusCodeError` if you don't

Reply:
yeah i worked my way through everything else, so that's the last system test to look at to verify that functionality/behavior